.github/workflows/npm-cd.yml

```diff
@@ -309,3 +309,117 @@ jobs:
               env:
                   NODE_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
+
+    test-release-candidate:
+        if: github.event_name != 'pull_request' && contains(github.ref, 'rc')
+        name: Test the release candidate
+        needs: publish-base-to-npm
+        runs-on: ${{ matrix.build.RUNNER }}
+        container:
+            image: ${{ matrix.build.IMAGE || '' }}
+            options: ${{ matrix.build.CONTAINER_OPTIONS || 'none'}}
+        strategy:
+            fail-fast: false
+            matrix:
+                build:
+                    - {
+                          RUNNER: ubuntu-latest,
+                      }
+                    - {
+                          RUNNER: [self-hosted, Linux, ARM64],
+                      }
+                    - {
+                          RUNNER: macos-12,
+                      }
+                    - {
+                          RUNNER: macos-13-xlarge,
+                      }
+                    - {
+                          RUNNER: [self-hosted, Linux, ARM64],
+                          IMAGE: 'node:alpine',
+                          CONTAINER_OPTIONS: '--user root --privileged --rm',
+                      }
+                    - {
+                          RUNNER: ubuntu-latest,
+                          IMAGE: 'node:alpine',
+                          CONTAINER_OPTIONS: '--user root --privileged',
+                      }
+        steps:
+            - name: install Redis, Python and git for alpine
+              if: ${{ matrix.build.IMAGE == 'node:alpine' }}
+              run: |
+                apk update
+                apk add redis python3 git
+                redis-server --daemonize yes
+            
+            - name: install Redis and Python for ubuntu
+              if: ${{ ! matrix.build.IMAGE == 'node:alpine' && ! contains(matrix.build.RUNNER, 'mac') }}
+              run: |
+                sudo apt-get update
+                sudo apt-get install redis-server python3
+                sudo systemctl start redis-server
+
+            - name: install Redis and Python for macos
+              if: ${{ contains(matrix.build.RUNNER, 'mac') }}
+              run: |
+                brew install redis python3
+                brew services start redis
+
+            # Currently "Checkout" action is not supported for musl on ARM64, so the checkout is happening on the runner and 
+            # here we just making sure we getting the clean repo
+            - name: Clean repository for musl on ARM64
+              if: ${{ contains(matrix.build.RUNNER, 'self-hosted') && matrix.build.IMAGE == 'node:alpine' }}
+              run: |
+                  git config --global --add safe.directory $GITHUB_WORKSPACE
+                  git clean -xdf
+                  git reset --hard
+                  git submodule sync
+                  git submodule update --init --recursive
+            
+            - name: Set up access for musl on ARM
```

Comment:
same as I wrote in the other PR - if you move all alpine setups to a separate action we can avoid the code duplication here too

Reply:
ping me when moved - i'll go back to review this file in this PR